### PR TITLE
Change Music datagen to read from file and add songs

### PR DESCRIPTION
**Pull request by ybyzek**

### CODE REVIEW

File: src/main/java/io/confluent/examples/streams/interactivequeries/kafkamusic/KafkaMusicExampleDriver.java [hunk 1]
```diff
@@ -57,69 +63,19 @@ public static void main(String [] args) throws Exception {
...
-                                                    "Hip Hop"),
-                                           new Song(12L,
-                                                    "The Calling",
-                                                    "Hilltop Hoods",
-                                                    "The Calling",
-                                                    "Hip Hop")
 
-    );
+    // Read comma-delimited file of songs into Array
+    List<Song> songs = new ArrayList<>();
```
Comment (mjsax) [inline src/main/java/io/confluent/examples/streams/interactivequeries/kafkamusic/KafkaMusicExampleDriver.java]: nit: add `final`

File: src/main/java/io/confluent/examples/streams/interactivequeries/kafkamusic/KafkaMusicExampleDriver.java [hunk 2]
```diff
@@ -57,69 +63,19 @@ public static void main(String [] args) throws Exception {
...
+    final String SONGFILENAME= "song_source.csv";
+    InputStream inputStream = KafkaMusicExample.class.getClassLoader().getResourceAsStream(SONGFILENAME);
```
Comment (mjsax) [inline src/main/java/io/confluent/examples/streams/interactivequeries/kafkamusic/KafkaMusicExampleDriver.java]: nit: add `final`

File: src/main/java/io/confluent/examples/streams/interactivequeries/kafkamusic/KafkaMusicExampleDriver.java [hunk 3]
```diff
@@ -57,69 +63,19 @@ public static void main(String [] args) throws Exception {
...
+    InputStreamReader streamReader = new InputStreamReader(inputStream, UTF_8);
```
Comment (mjsax) [inline src/main/java/io/confluent/examples/streams/interactivequeries/kafkamusic/KafkaMusicExampleDriver.java]: nit: add `final`

File: src/main/java/io/confluent/examples/streams/interactivequeries/kafkamusic/KafkaMusicExampleDriver.java [hunk 4]
```diff
@@ -57,69 +63,19 @@ public static void main(String [] args) throws Exception {
...
+    BufferedReader br = new BufferedReader(streamReader);
```
Comment (mjsax) [inline src/main/java/io/confluent/examples/streams/interactivequeries/kafkamusic/KafkaMusicExampleDriver.java]: nit: add `final`

File: src/main/java/io/confluent/examples/streams/interactivequeries/kafkamusic/KafkaMusicExampleDriver.java [hunk 5]
```diff
@@ -57,69 +63,19 @@ public static void main(String [] args) throws Exception {
...
+    for (String line; (line = br.readLine()) != null;) {
```
Comment (mjsax) [inline src/main/java/io/confluent/examples/streams/interactivequeries/kafkamusic/KafkaMusicExampleDriver.java]: nit: add `final`

File: src/main/java/io/confluent/examples/streams/interactivequeries/kafkamusic/KafkaMusicExampleDriver.java [hunk 6]
```diff
@@ -57,69 +63,19 @@ public static void main(String [] args) throws Exception {
...
+      String[] values = line.split(",");
```
Comment (mjsax) [inline src/main/java/io/confluent/examples/streams/interactivequeries/kafkamusic/KafkaMusicExampleDriver.java]: nit: add `final`

File: src/main/java/io/confluent/examples/streams/interactivequeries/kafkamusic/KafkaMusicExampleDriver.java [hunk 7]
```diff
@@ -57,69 +63,19 @@ public static void main(String [] args) throws Exception {
...
+      Song newSong = new Song(Long.parseLong(values[0]),values[1],values[2],values[3],values[4]);
+      songs.add(newSong);
+    }
+    br.close();
```
Comment (mjsax) [inline src/main/java/io/confluent/examples/streams/interactivequeries/kafkamusic/KafkaMusicExampleDriver.java]: We should use a try-with-resources to ensure it gets closed.

File: src/main/java/io/confluent/examples/streams/interactivequeries/kafkamusic/KafkaMusicExampleDriver.java (hunk 1 above)
Comment (ybyzek) [reply src/main/java/io/confluent/examples/streams/interactivequeries/kafkamusic/KafkaMusicExampleDriver.java]: I added all the `final`

File: src/main/java/io/confluent/examples/streams/interactivequeries/kafkamusic/KafkaMusicExampleDriver.java (hunk 7 above)
Comment (ybyzek) [reply src/main/java/io/confluent/examples/streams/interactivequeries/kafkamusic/KafkaMusicExampleDriver.java]: I reworked the code...how is rev 2?

File: src/main/java/io/confluent/examples/streams/interactivequeries/kafkamusic/KafkaMusicExampleDriver.java [hunk 8]
```diff
@@ -57,69 +65,30 @@ public static void main(String [] args) throws Exception {
...
-
-    );
+
+    // Read comma-delimited file of songs into Array
+    final List<Song> songs = new ArrayList<>();
+    final String SONGFILENAME= "song_source.csv";
+    final InputStream inputStream = KafkaMusicExample.class.getClassLoader().getResourceAsStream(SONGFILENAME);
+    final InputStreamReader streamReader = new InputStreamReader(inputStream, UTF_8);
+    final BufferedReader br = new BufferedReader(streamReader);
+    try {
```
Comment (mjsax) [inline src/main/java/io/confluent/examples/streams/interactivequeries/kafkamusic/KafkaMusicExampleDriver.java]: Nit: update to "try-with-resource" clause:
```
try (final BufferedReader br = new BufferedReader(streamReader)) {
  ...
}
```
This allows to omit `catch` and `final` block .

File: src/main/java/io/confluent/examples/streams/interactivequeries/kafkamusic/KafkaMusicExampleDriver.java [hunk 9]
```diff
@@ -57,69 +65,30 @@ public static void main(String [] args) throws Exception {
...
+      String line = null;
+      while ((line = br.readLine()) != null) {
+        final String[] values = line.split(",");
+        Song newSong = new Song(Long.parseLong(values[0]),values[1],values[2],values[3],values[4]);
```
Comment (mjsax) [inline src/main/java/io/confluent/examples/streams/interactivequeries/kafkamusic/KafkaMusicExampleDriver.java]: nit: add `final` and add spaces after commas.

**Comment by ybyzek** [comment]
Thank you for your help @mjsax .  Now that it is committed into `4.0.0-post`, do I need to run `pint merge`?  Or will it automatically propagate to `4.1.x`?

**Pull request merged**